### ISSUE TITLE
Health info misleading after health revamp, and junk food intolerance has become free points

### ISSUE THREAD

**Issue by PatrikLundell** [CleverRaven/Cataclysm-DDA#59505]
### Describe the bug

After a recent health rework "junk food" is now very harmful to PC health, but the UI field that's intended to warn against harmful food does not reflect that, so a player has to firstly know that "junk food" is directly harmful, and then check each food item to check whether it's labeled as such (salted popcorn is junk food, while unsalted popcorn isn't).

Also, the change makes junk food intolerance a free point pick, as you can't eat junk food anyway.

### Steps to reproduce

1. Upgrade the game to a recent version.
2. Continue to eat whatever has the shortest expiry date that isn't marked as harmful.
3. Get plagued by morning messages that are caused by bad heath, plummeting into very bad health within a week, as revealed by searching for the meaning of those obscure messages.
4. Try to compensate with vitamin containing food and vitamin pills.
5. Note that the inherent junk food danger isn't affected by vitamins. Sure, the enormous calorie consumption ought to have made the player put on weight rapidly, but that's required by the game not to lose weight.
6. Be surprised that most of the worlds population hasn't died yet due to the incredibly dangerous junk food.

### Expected behavior

If a food item has a significant negative health effect, the health column should indicate that. All junk food should have at least one bar, and probably two.

Remove the junk food intolerance trait, as junk food is now too harmful for anyone to eat (possibly with the exception of some iron guts trait or some mutations). It may be somewhat safe to eat at the same level as tainted meat presumably is.

### Versions and configuration

- OS: Windows
    - OS Version: 10.0.19043.1826 (21H1)
- Game Version: d6a1777 [64-bit]
- Graphics Version: Tiles
- Game Language: System language []
- Mods loaded: [
    Dark Days Ahead [dda],
    Disable NPC Needs [no_npc_food],
    No Fungal Growth [no_fungal_growth],
    Bionic Professions [package_bionic_professions]
]

**Comment by anoobindisguise** [comment]
Not all junk food is unhealthy. Chocolate bars, cookies, cakes, sports drinks (gatorade I think), chocolate drinks, pretzels (chocolate covered pretzels), pudding, marshmallows, mayo, mustard, tortilla chips, toastems, pork sticks and potato chips are all things my Sweet Toothed survivor eats on the regular for the mood boost and they do not have (-) health, although I have noticed that despite chugging gamma globulin shots my health is still pretty poor. is there a separate mechanic that's causing all junk food to suck my health away?

**Comment by PatrikLundell** [comment]
My experience indicates the changes made has turned all junk food into toxic health hazards. My character's diet consisted of things like cookies, toastems, marshmallows, popcorn, etc. plus various chocolate covered things, etc. The health tanked like a stone after upgrading. After tossing out all junk food (into the base calorie store) my character's health modifier (debug info) has turned positive, but the health stat isn't recovering: it has just stopped sinking at -108. There may be some hidden hysteresis variable that might cause a positive modifier to not actually result in a positive change until inertia has stopped the drop, or there may be some bug at play.

I suspect Sweet Tooth is now shot, though, converted into a useless waste of points, and yes, I think all junk food is now toxic.

**Comment by Terrorforge** [comment]
I'm not sure this is quite what's happening, but health history is essentially stored twice. daily_health is added to health_tally every day. but daily_health is never reset, it just decays slowly. This makes it incredibly hard to reverse a health shift in either direction, because you have to first overcome daily_health, which will only then start incrementing health_tally, which will then start moving the actual health value.

I've been unsure whether to report this as a bug, because being resistant to change is an intended feature of the new health system.

**Comment by PatrikLundell** [comment]
I'm completely fine with health being resistant to change, but it's completely wrong to try to implement that by applying all the resistance to the health trend. A better way would probably be to make the maximum health trend value quite small compared to the overall health value so daily changes would only make small dents, and I might consider major health threats to have their own negative factors so you can e.g. very rapidly get radiation poisoning, but it takes a long time to recover from it.
I also find it odd that junk food should essentially contain large amounts of some nebulous health poison.

**Comment by chrispikula** [comment]
The delay of player action -> player health is a bit hard to understand via gameplay.  The phase delay doesn't appear to have 

Perhaps, if health_tally was to have some sort of indication in the log if you were abusing it, it would be nice?  Things like the following output messages:

--
"Your skin feels greasy after eating all that junk food."

-
"Your stomach rumbles after eating all that processed food."

+
"Your skin glows with natural vigor, it must be your diet."

++
"You have a pleasant dream about the tasty food you ate yesterday."

Getting an indication that junk food is really bad for you with the -- message would be nice.

**Comment by Monopool** [comment]
Considering that no food is considered inherently healthy anymore, I don't think it makes sense to arbitrarily consider (most) junk food inherently unhealthy anymore. Stuff that poses more immediate health risks being the exception (ex. atomic energy drinks/coffee, methacola, dangerous chemicals, mutagen sources, mild untracked toxins like raw potatoes, etc.)

Most junk food is considered unhealthy because it is a large amount of empty calories without nutrients, or has a lot of fat or sugar. However, this is in no way consistently enforced under the current system and we have systems to regulate health based on weight and vitamin intake anyway.

Ice cream is probably the worst offender since chocolate ice cream is suddenly unhealthy when crafted despite all of the ingredients lacking a penalty. To top it off, chocolate bars, chocolate milk, **milkshakes**, and shelf stable chocolate drink are all fine.

Somehow mixing and freezing ingredients that are all safe to consume separately makes them **directly** harm your health, yet curiously you can consume an entire jar of mayonnaise, a jug of cooking oil, pounds of pure sugar, bottles of maple syrup, etc. with no direct health penalty.

In fact, I'm pretty sure the game considers appropriate rations of pure mayo and multivitamin pills to be "healthy" unless I'm missing something.

tl:dr: I think (most) junk food needs to be changed to no longer be directly unhealthy or be balanced with the reintroduction of directly "healthy" foods. Let the excessive calories and lack of vitamins speak for themselves under the new system.

**Comment by kevingranade** [comment]
Helping stalebot catch up.
Closing vague characterization of problem.
